src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<T> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            machines = new BulldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            machines = new ExcavatorProducer().get();
+        }
+        if (type == Truck.class) {
+            machines = new TruckProducer().get();
+        }
+        return (List<T>) machines;
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, T value) {
+        int size = machines.size();
+        for (int i = 0; i < size; i++) {
+            machines.set(i, value);
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+        if (!machines.isEmpty()) {
+            for (Machine machine : machines) {
+                machine.doWork();
+            }
+        }
```

Comment:
idk , its an individual case if u want co check machine list is not empty
If you think it's necessary, okay, I just think it won't affect anything as it is.